doc/source/tutorial/examples/signal_SpectralAnalysis_SpectrumPhaseDelay.py

```diff
@@ -0,0 +1,41 @@
+import matplotlib.gridspec as gridspec
+import matplotlib.pyplot as plt
+import numpy as np
+
+from scipy import signal
+from scipy.fft import rfft, rfftfreq
+
+# Create input signal:
+n = 50
+x = np.zeros(n)
+x[0] = n
+
+# Apply FIR filter which delays signal by 3 samples:
+y = signal.lfilter([0, 0, 0, 1], 1, x)
+
+X, Y = (rfft(z_) / n for z_ in (x, y))
+f = rfftfreq(n, 1)  # sampling interval T = 1 s
+
+fig = plt.figure(tight_layout=True, figsize=(6., 4.))
+gs = gridspec.GridSpec(3, 1)
+ax0 = fig.add_subplot(gs[0, :])
+ax1 = fig.add_subplot(gs[1:, :], sharex=ax0)
+
+for Z_, n_, m_ in zip((X, Y), ("Input $X(f)$", "Output $Y(f)$"), ('+-', 'x-')):
+    ax0.plot(f, abs(Z_), m_, alpha=.5, label=n_)
+    ax1.plot(f, np.unwrap(np.angle(Z_)), m_, alpha=.5, label=n_)
+
+ax0.set(title="Impulse Response of 3 Sample Delay Filter (no window)",
+        ylabel="Magnitude", xlim=(0, f[-1]), ylim=(0, 1.1))
```

Comment:
To me "impulse response" means the x axis should be time and "frequency response" the x axis should be frequency. I was really confused when I read this axis's title and saw all ones when I expected three zeros followed by a single one for the 3-sample delay filter...
```suggestion
ax0.set(title="Frequency Response of 3 Sample Delay Filter (no window)",
        ylabel="Magnitude", xlim=(0, f[-1]), ylim=(0, 1.1))
```
If you do want to keep an impulse response I think you should add a third axis at the top that shows the time waveforms